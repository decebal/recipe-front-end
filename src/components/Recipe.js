// @flow

import React from 'react';
import PropTypes from 'prop-types';
import {
  Button, ButtonGroup,
  Container,
  H2, Hr,
  Jumbotron,
  ListGroup,
  ListGroupItem,
  P,
  Row,
} from '@bootstrap-styled/v4/dist/@bootstrap-styled/v4';
import * as uuid4 from 'uuid4';
import { Headline } from './layout';
import type RecipeType from '../types/recipe';


const listItems = ({ items }) => Object.values(items).map((item, key) => (
  <ListGroupItem key={uuid4()} style={{ textAlign: 'left' }}>
    {key + 1}
    {' . '}
    {item}
  </ListGroupItem>
));

const Recipe = ({
  id, name, description, ingredients, instructions, onDelete,
}: RecipeType) => (
  <div>
    <Headline>{name}</Headline>
    <Row>
      <H2>Description</H2>
    </Row>
    <Jumbotron fluid>
      <Container>
        <P className="lead" style={{ textAlign: 'left' }}>{description}</P>
      </Container>
    </Jumbotron>
    <Hr />
    <Row>
      <H2>Ingredients</H2>
    </Row>
    <ListGroup>
      { listItems({ items: ingredients }) }
    </ListGroup>
    <Hr />
    <Row>
      <H2>Instructions</H2>
    </Row>
    <ListGroup>
      { listItems({ items: instructions }) }
    </ListGroup>
    <Hr />
    <Row style={{ flexFlow: 'row-reverse' }}>
      <ButtonGroup>
        <Button outline color="warning" href={`/recipe/${id}/edit`}>Edit Recipe</Button>
        <Button outline color="danger" onClick={() => onDelete({ id })}>Delete Recipe</Button>
      </ButtonGroup>
    </Row>
    <Hr />
  </div>
);

Recipe.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.string).isRequired,
  instructions: PropTypes.arrayOf(PropTypes.string).isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default Recipe;
