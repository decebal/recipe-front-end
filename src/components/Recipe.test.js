import React from 'react';
import renderer from 'react-test-renderer';
import Recipe from './Recipe';


test('Recipe renders correctly', () => {
  const tree = renderer
    .create(<Recipe
      name=""
      description=""
      ingredients={[]}
      instructions={[]}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
