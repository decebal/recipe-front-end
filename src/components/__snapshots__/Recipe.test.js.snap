// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders correctly 1`] = `
<div>
  <h1
    className="sc-bdVaJa qBDhK"
  >
    
  </h1>
  <div
    className="Row-sc-2hg243-0 gRqPSy row"
  >
    <h2
      className="H2-kplljn-0 bPmkUD"
    >
      Description
    </h2>
  </div>
  <div
    className="Jumbotron-sc-4btr39-0 chpogd jumbotron jumbotron-fluid"
  >
    <div
      className="Container-u38a83-0 jDuhNh container"
    >
      <p
        className="lead P-sc-31hrrf-0 ivKXeO"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        
      </p>
    </div>
  </div>
  <hr
    className="Hr-sc-1a12p8r-0 jPQocB"
  />
  <div
    className="Row-sc-2hg243-0 gRqPSy row"
  >
    <h2
      className="H2-kplljn-0 bPmkUD"
    >
      Ingredients
    </h2>
  </div>
  <ul
    className="ListGroup-sc-1ugap33-0 ewIJD list-group Ul-sctud2-0 ldzbsT"
    role="navigation"
    tag="nav"
    toggleable="false"
  />
  <hr
    className="Hr-sc-1a12p8r-0 jPQocB"
  />
  <div
    className="Row-sc-2hg243-0 gRqPSy row"
  >
    <h2
      className="H2-kplljn-0 bPmkUD"
    >
      Instructions
    </h2>
  </div>
  <ul
    className="ListGroup-sc-1ugap33-0 ewIJD list-group Ul-sctud2-0 ldzbsT"
    role="navigation"
    tag="nav"
    toggleable="false"
  />
  <hr
    className="Hr-sc-1a12p8r-0 jPQocB"
  />
  <div
    className="Row-sc-2hg243-0 gRqPSy row"
    style={
      Object {
        "flexFlow": "row-reverse",
      }
    }
  >
    <div
      className="ButtonGroup-sc-1wk454k-0 euUImV btn-group"
      role="group"
    >
      <a
        className="Button-h8watj-0 OlpfH btn btn-outline-warning"
        href="/recipe/invalid-uuid-4/edit"
        onClick={[Function]}
      >
        Edit Recipe
      </a>
      <button
        className="Button-h8watj-0 OlpfH btn btn-outline-danger"
        onClick={[Function]}
        type="button"
      >
        Delete Recipe
      </button>
    </div>
  </div>
  <hr
    className="Hr-sc-1a12p8r-0 jPQocB"
  />
</div>
`;
